universal-application-tool-0.0.1/app/auth/UATProfile.java

```diff
@@ -1,11 +1,46 @@
 package auth;
 
+import com.google.common.base.Preconditions;
+import java.time.Clock;
+import models.Account;
+import models.Applicant;
 import org.pac4j.core.profile.CommonProfile;
 
 public class UATProfile extends CommonProfile {
+  private final Clock clock;
+  private Applicant applicant;
+
   public UATProfile() {
+    this(Clock.systemDefaultZone());
+  }
+
+  public UATProfile(Clock clock) {
     super();
-    // TODO: write this to the database, read the ID back, and use that as the ID here.
-    setId("TODO");
+    this.clock = clock;
+  }
+
+  public Applicant getApplicant() {
+    if (applicant == null) {
+      this.applicant = new Applicant();
+      this.applicant.id = Long.valueOf(this.getId());
+      this.applicant.refresh();
+    }
+    return applicant;
+  }
```

Comment:
Question (blocking): will this code run on every request where we need to access the requester's `Applicant` or `Account`? If so it should be async so it doesn't block HTTP threads on the applicant code paths.

Reply:
Sure thing.